File: src/main/java/org/example/TC/삼총사.java
package org.example.TC;

import java.util.ArrayList;
import java.util.List;

public class 삼총사 {
    public int solution(int[] number) {
        int count = 0;
        for(int i = 0; i<number.length; i++ ) {
            for(int j=0; j<i; j++) {
                for(int k=0; k<j; k++) {
                    if(number[i]+number[j] == -1*number[k]){
                        count++;
                    }
                }
            }
        }
        return count;
    }
}

File: src/main/java/org/example/TCK/삼총사.kt
package org.example.TCK
/*https://school.programmers.co.kr/learn/courses/30/lessons/131705*/
class 삼총사 {
    fun solution(number:Array<Int>):Int{
        var answer:Int=0
        for((i,n) in number.withIndex()) { /*withIndex()를 사용하면(index, value) 쌍 값을 얻을 수 있다. */
            for((j,m) in number.withIndex()) {
                for((k,l) in number.withIndex()) {
                    if((i>j)&&(j>k)) {
                        if(n+m+l==0) {
                            answer++
                        }
                    }
                }
            }
        }

        return answer;
    }
}
